films/repository/film/repo_film.go

```diff
@@ -0,0 +1,246 @@
+package film
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+	"time"
+
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/configs"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/pkg/models"
+	"github.com/lib/pq"
+
+	_ "github.com/jackc/pgx/stdlib"
+)
+
+type IFilmsRepo interface {
+	GetFilmsByGenre(genre uint64, start uint64, end uint64) ([]models.FilmItem, error)
+	GetFilms(start uint64, end uint64) ([]models.FilmItem, error)
+	GetFilm(filmId uint64) (*models.FilmItem, error)
+	GetFilmRating(filmId uint64) (float64, uint64, error)
+	FindFilm(title string, dateFrom string, dateTo string,
+		ratingFrom float32, ratingTo float32, mpaa string, genres []string, actors []string) ([]models.FilmItem, error)
```

Comment:
Я бы сделал так - несколько проще понят что функция вернет, да и выглядит не так громоздко
```suggestion
	FindFilm(title string, dateFrom string, dateTo string,
		ratingFrom float32, ratingTo float32, mpaa string, genres []string, actors []string,
	) ([]models.FilmItem, error)
```